Objects/exceptions.c

```diff
@@ -2847,7 +2847,10 @@ _set_legacy_print_statement_msg(PySyntaxErrorObject *self, Py_ssize_t start)
     // PRINT_OFFSET is to remove `print ` word from the data.
     const int PRINT_OFFSET = 6;
     Py_ssize_t text_len = PyUnicode_GET_LENGTH(self->text);
-    PyObject *data = PyUnicode_Substring(self->text, PRINT_OFFSET, text_len);
+    // Issue 32028: Handle case when whitespace is used with print call
+    PyObject *initial_data = _PyUnicode_XStrip(self->text, 2, strip_sep_obj);
```

Comment:
Rather than adding a new call to `_PyUnicode_XStrip`, just reorder the existing operations:

1. First strip any surrounding ASCII whitespace (i.e. do this first, instead of last)
2. Then get the length of the result (rather than the length of the original text)
3. Then skip over `PRINT_OFFSET` characters at the beginning

Reply:
A minor point I missed when reviewing the original PR: defining and using `const int STRIP_BOTH = 2;` will make the `_PyUnicode_XStrip` call more self explanatory.

So we may as well make that change now, since the code is being modified anyway.

Reply:
+1 for the `const int STRIP_BOTH = 2;`. I was explaining my patch to a few folks and I had to consult my blog for that particular thing. Few months down the line I didn't even remember that.

So, yes that would be great.

Reply:
@ncoghlan So, while I modified the code, one of the test cases fails with re-ordering the code. That test case is:

`test_string_with_excessive_whitespace`. Since we're only stripping the the data at the beginning, and then directly trying to extract the sub-string.

I think we may need to strip the leading chars to make the previous test case pass. What do you say?

cc @serhiy-storchaka

Reply:
I've pushed my current code which re-orders the existing operations & the `test_string_with_excessive_whitespace` is failing. I'm waiting for your reply on this. I think we might want to strip the leading whitespace from `data`.